### DISCUSSION TITLE
Updating DataGrid not as expected

### DISCUSSION THREAD

**Discussion by MichelvanVuren** [AvaloniaUI/Avalonia#6487] [Q&A] [answered]
I want to update a DataGrid with the use of a button. I can see the change of the collection in the sorting of the DataGrid but the new value of the field isn't shown in the view of the DataGrid.
I know I can use a ListBox or use the direct editing of the DataGrid, but still:
Why isn't the value updated? Any ideas?

I've made a small example in: https://github.com/MichelvanVuren/AvaloniaTest
![AvaloniaTest-2021-08-28-08-24-21_EDIT](https://user-images.githubusercontent.com/75127871/131229493-c5740b35-5553-4be8-9c62-0d99d76dee31.gif)
The delete-button is acting as expected.

**Comment by jp2masa** [comment, marked as answer]
I think it's because `Country` should be a view model and implement `INotifyPropertyChanged`.

An alternative would be to make `Country` immutable and replace [these lines](https://github.com/MichelvanVuren/AvaloniaTest/blob/f85cc485d5f487a81a2b28268816fa533f7d50a3/ViewModels/DataGridEditingViewModel.cs#L80-L81) with:

```cs
SelectedCountry = new Country() { CountryId = SelectedCountry.CountryId, Name = Input };
_countryCache.AddOrUpdate(SelectedCountry);
```

Or with records:

```cs
public record Country(int CountryId, string Name);
```

And replace the other 2 lines with:

```cs
SelectedCountry = SelectedCountry with { Name = Input };
_countryCache.AddOrUpdate(SelectedCountry);
```

**Reply by MichelvanVuren** [reply]
Thank you. I've updated my test with one of your solutions and it works!
